### ISSUE TITLE
About training the ranker

### ISSUE THREAD

**Issue by yhshu** [salesforce/rng-kbqa#19]
https://github.com/salesforce/rng-kbqa/blob/2b6ef28e7724f11181f59589398894a1d0617455/GrailQA/scripts/run_ranker.sh#L45

Hi,
I would like to ask the batch size of the ranker (not entity disambiguation).
In the paper, the batch size is 8. However, the script here is 1.

Besides, when evaluating (predicting) the LF ranking, should the batch size be 1 according to BERTCandidateRanker:
https://github.com/salesforce/rng-kbqa/blob/2b6ef28e7724f11181f59589398894a1d0617455/framework/models/BertRanker.py#L30

But the script set the evaluation batch size to 128 as shown in,
https://github.com/salesforce/rng-kbqa/blob/2b6ef28e7724f11181f59589398894a1d0617455/GrailQA/scripts/run_ranker.sh#L47

Are these two numbers have different meanings?
Would you please provide me with a clue? Thanks a lot.

**Comment by xiye17** [comment]
In the paper, the batch size is 8. However, the script here is 1.
----------
the batch size 1 is per gpu. We have 4 gpu and we do gradient accumulation as 2 (line 46 in run_ranker.sh). So in total this is batch size 8 (4 * 2).

But the script set the evaluation batch size to 128 as shown in
----------
When doing inference, for each query, we'll need to score thousands of logical forms. And we batch them so we score 128 per gpu. Here the batch size 1 means this 128 logical forms are all for a single query. Sorry for the confusing comments.

**Comment by yhshu** [comment]
Thanks, got it. 
But to be more clear, about, the training batch size, does it refer to 1 query and 2 candidate LFs for each batch, or 2 queries and all their candidates (100 samples including up to 96 negative ones) for each batch?
I guess the eval batch size means 1 query and 128 candidate LFs.

**Comment by xiye17** [comment]
batch size 8 means 8 questions  (containing 8 gold logical forms and 8*96 negative examples)

**Comment by yhshu** [comment]
Understood, thanks.
